### ISSUE TITLE
Increase tolerance for is_valid()?

### ISSUE THREAD

**Issue by saurabh02** [materialsproject/pymatgen#386]
## Summary

* Should the tolerance distance for SiteCollection.is_valid() in pymatgen/pymatgen/core/structure.py be higher than the default 0.01 A? 

## Example

For a H atom, atomic_radius = 0.25 A, atomic_radius_calculated = 0.53 A and van der Waal's radius = 1.2 A. I have a structure that has atoms within 0.05 A which should be invalid. 

## Suggested solution (if any)

Increase tolerance to 0.75 A?

**Comment by shyuep** [comment]
Can you just implement a tolerance factor to be added as an arg. Thanks.

**Comment by saurabh02** [comment]
It is already an argument, but with a default of 0.01 A. Should the default be increased?

**Comment by shyuep** [comment]
No. You set the value to something you need. H is a special case.


Shyue Ping

On May 19, 2016 at 5:45:03 PM, Saurabh Bajaj (notifications@github.com) wrote:

It is already an argument, but with a default of 0.01 A. Should the default be increased?

**Comment by saurabh02** [comment]
Okay thanks

**Comment by computron** [comment]
Let me just a bit more context here -

Saurabh and I parsed the Pauling file database, which is full of bad structures for various reasons. We relied on the default "is_valid()" method to help in avoiding some of the bad structures. However, we noticed that there are many bad structures that pass through this method, i.e., structures where O-O bonds are 0.2 Angstroms in length. We feel that:

* <0.01 Angstrom atom distance = certainly wrong structure, and is_valid() correctly marks as false
* 0.01-0.5 Angstrom distance = also certainly wrong structure, but default is_valid() marks as valid. We are saying *certainly* wrong because the smallest bond distance should be about 0.73 Angstrom which is a hydrogen-hydrogen bond. There will be no smaller bonds than that. Taking a little bit of safety tolerance, we reduce that down to 0.5 Angstrom (Saurabh wrote 0.75 Angstrom, but 0.5 Angstrom is probably a safer bet to ensure that *zero* valid structures pass through)
* 0.5 Angstrom - 0.72 Angstrom - extremely likely to be invalid structure, but maybe be on the cautious side and have default is_valid() mark them as valid (retain current behavior)
* >0.72 Angstrom - should be valid, i.e. retain current behavior

Basically, we are saying there is significant room to crank up the default tolerance for is_valid() such that *zero* valid structures are filtered out, while doing a better job of filtering out invalid structures. Of course, if there are any valid structures with atom distances in the 0.01 Angstrom - 0.5 Angstrom range, this is null and void. We just don't think are any.
